packages/deployments/contracts/contracts/lib/Connext/AssetLogic.sol

```diff
@@ -24,20 +26,22 @@ library AssetLogic {
   function transferAssetToContract(
     address _assetId,
     uint256 _specifiedAmount,
+    uint256 _additionalNativeAmount,
```

Comment:
what this function is doing feels a bit unintuitive -- im wondering if we should rename it or break it up. by the name it presumably just transfers an asset to the contract. instead it seems a catch-all method for transferring to the contract, wrapping native asset, and transferring some additional native asset that doesn't get wrapped.

perhaps it would be better to have a vanilla `transferAssetToContract`, `transferAndWrapNativeAsset`, `transferNativeAsset`

not 100% sure about it though so open to other feedback here

Reply:
I do agree with the idea of this function name is unintuitive, also I think we could have those 3 functions but I would  leave the logic to decide which on to use in this function rather to move it to the `execute`

Reply:
Refactored the method, changed the name and split the logic into other methods https://github.com/connext/nxtp/pull/955/commits/3852d8958fb169c945cf90246c94e0fa58de2051

Reply:
awesome looks much cleaner!